test/case/ietf_system/add_delete_user.py

```diff
@@ -15,6 +15,12 @@ def generate_restrictred_credential():
 
     return credential
 
+def generate_password_hash(password):
+    command = f"mkpasswd -m SHA-256 {password}"
```

Comment:
This was a clever idea!  Was expecting a python replacement, but this will work just fine too.  Just to clarify, this is `mkpasswd` from within the container, so the argument to the `-m` option should be one of: `md5`, `sha256`, or `sha512`.  Other versons of the mkpasswd too exist, but this one is the same as we have in Infix (from the BusyBox tool suite).

I suggest adding two arguments to the Python method:
 1. `type="md5"`, argument with default value
 2. `salt`, a random string. If the user provides it, append it to the end of the `command` before running it

Reply:
It may look like a clever idea, but I felt I have no other choice :(
I spent too much time trying to find a python function which would do exactly the same thing as "crypt", but at the end none of it worked for me. At least couldn't make it using: bcrypt, passlib, hashlib, argon2 and cryptography.
When it comes to passlib and argon2 at least, I couldn't even get it to the environment properly (install + import weren't enough just enough)... If I knew how to import it properly, maybe that one would work...

Reply:
> [...] I couldn't even get it to the environment properly (install + import weren't enough just enough)... If I knew how to import it properly, maybe that one would work...

Sorry to hear that!  I'm guessing there's a few libraries missing from the python install in the [docker container](https://github.com/kernelkit/infix/blob/main/test/docker/Dockerfile).  For future reference, of if you want to just know how one can update the container: see the [test/docker/README](https://github.com/kernelkit/infix/blob/main/test/docker/README.md)

Reply:
> Thnx for the comments, and if you think it is important to have a python function here, we can spend some more time. Otherwise, I'll just make an update as you proposed in the comments :)

This is a perfectly fine solution, and also verifies the same function (mkpasswd) that we use in Infix.  You definitely don't have to spend any more frustrating hours on it! <3